components/script/script_thread.rs

```diff
@@ -854,19 +860,35 @@ impl ScriptThreadFactory for ScriptThread {
     }
 }
 
+#[derive(Debug)]
+pub struct LayoutNotFoundError {
+    pipeline_id: PipelineId,
+}
+
+impl LayoutNotFoundError {
+    fn new(pipeline_id: PipelineId) -> Self {
+        Self { pipeline_id }
+    }
+}
+
+impl std::fmt::Display for LayoutNotFoundError {
+    fn fmt(&self, format: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(format, "No layout found for {}", self.pipeline_id)
+    }
+}
+
```

Comment:
Please hold off on doing this for now. This clippy issue should be addressed in a different way.